viadot/sources/cloud_for_customers.py

```diff
@@ -5,56 +5,169 @@
 from urllib.parse import urljoin
 from ..config import local_config
 
+import time
+from requests.adapters import HTTPAdapter
+from requests.exceptions import ConnectionError, HTTPError, ReadTimeout, Timeout
+from urllib3.exceptions import ProtocolError
+import logging
+import re
+from requests.packages.urllib3.util.retry import Retry
 
-class CloudForCustomers(Source):
-    """
-    Fetches data from Cloud for Customer.
+url = "https://my341115.crm.ondemand.com/sap/c4c/odata/ana_businessanalytics_analytics.svc/$metadata?entityset=RPZ1F90A299C830A29C6659D3QueryResult"
+
+
+def column_mapper(url=url):
+    credentials = local_config.get("CLOUD_FOR_CUSTOMERS")
+    access = credentials.get("Prod")
+    auth = (access["username"], access["password"])
+    response = requests.get(url, auth=auth)
+    column_mapping = {}
+    for sentence in response.text.split("/>"):
+        result = re.search(r'(?<=Name=")([^"]+).+(sap:label=")([^"]+)+', sentence)
+        if result:
+            key = result.groups(0)[0]
+            val = result.groups(0)[2]
+            column_mapping[key] = val
+    return column_mapping
+
+
+def change_to_meta_url(url):
+    meta_url = ""
+    start = url.split(".svc")[0]
+    ending = url.split("/")[-1]
+    e = ending.split("?")[0]
```

Comment:
What's e? This should either be a better variable name or explained in the function's docstrings.